flash/core/registry.py

```diff
@@ -0,0 +1,99 @@
+from collections import defaultdict
+from functools import partial
+from types import FunctionType
+from typing import Callable, Dict, Mapping, Optional, Union
+
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch.nn import Module
+
+
+class FlashRegistry(Dict):
+    """
+    This class is used to register function or partial to a registry:
+
+    Example::
+
+        backbones = FlashRegistry("backbones")
```

Comment:
Where will the registry instance live? Once per task?

Reply:
Yes, possible or 1 for all.